pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -259,17 +260,28 @@ func (r *ReconcileKnativeServing) checkDeployments(instance *servingv1alpha1.Kna
 // Check for all dependencies
 func (r *ReconcileKnativeServing) checkDependencies(instance *servingv1alpha1.KnativeServing) error {
 	defer r.updateStatus(instance)
-
-	istio := schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "virtualservice"}
-	list := &unstructured.UnstructuredList{}
-	list.SetGroupVersionKind(istio)
-	if err := r.client.List(context.TODO(), nil, list); err != nil {
+	checkError := func(err error, instance *servingv1alpha1.KnativeServing) error {
 		msg := fmt.Sprintf("Istio not detected; please install ServiceMesh")
 		instance.Status.MarkDependencyMissing(msg)
 		log.Error(err, msg)
 		return err
 	}
-
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: "istio-system", Name: "default"}, smmr); err != nil {
+		return checkError(err, instance)
+	}
+	found := func(smmr *maistrav1.ServiceMeshMemberRoll) bool {
+		for _, member := range smmr.Status.ConfiguredMembers {
+			if member == "knative-serving" {
+				return true
+			}
+			return false
```

Comment:
I don't think this return should be here. It means the function will only ever return true if `knative-serving` is the first member of the list. If it's not, the loop should continue. Only after it loops through all members without finding a match should it return false.